src/EditorFeatures/Core/InlineRename/AbstractEditorInlineRenameService.cs

```diff
@@ -30,154 +21,14 @@ protected AbstractEditorInlineRenameService(IEnumerable<IRefactorNotifyService>
-                {
-                    var solution = document.Project.Solution;
-                    var sourceDocument = solution.GetRequiredDocument(location.SourceTree);
-
-                    if (sourceDocument is SourceGeneratedDocument)
-                    {
-                        // The file is generated so we can't go editing it (for now)
-                        return new FailureInlineRenameInfo(EditorFeaturesResources.You_cannot_rename_this_element);
-                    }
-
-                    if (document.Project.IsSubmission)
-                    {
-                        var projectIdOfLocation = sourceDocument.Project.Id;
-
-                        if (solution.Projects.Any(p => p.IsSubmission && p.ProjectReferences.Any(r => r.ProjectId == projectIdOfLocation)))
-                            return new FailureInlineRenameInfo(EditorFeaturesResources.You_cannot_rename_elements_from_previous_submissions);
-                    }
-                    else
-                    {
-                        // We eventually need to return the symbol locations, so we must convert each location to a DocumentSpan since our return type is language-agnostic.
-                        documentSpans.Add(new DocumentSpan(sourceDocument, location.SourceSpan));
-                    }
-                }
-                else
-                {
-                    return new FailureInlineRenameInfo(EditorFeaturesResources.You_cannot_rename_this_element);
-                }
-            }
-
-            var sourceText = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
-            var triggerText = sourceText.ToString(triggerToken.Span);
-            var fallbackOptions = _globalOptions.CreateProvider();
+            var symbolicInfo = await SymbolicRenameInfo.GetRenameInfoAsync(document, position, cancellationToken).ConfigureAwait(false);
+            if (symbolicInfo.LocalizedErrorMessage != null)
+                return new FailureInlineRenameInfo(symbolicInfo.LocalizedErrorMessage);
 
             return new SymbolInlineRenameInfo(
-                refactorNotifyServices, document, triggerToken.Span, triggerText,
-                symbol, forceRenameOverloads, documentSpans.ToImmutableAndFree(),
-                fallbackOptions, cancellationToken);
-        }
-
-        private static async Task<SyntaxToken> GetTriggerTokenAsync(Document document, int position, CancellationToken cancellationToken)
-        {
-            var syntaxTree = await document.GetRequiredSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
-            var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
-            var token = await syntaxTree.GetTouchingWordAsync(position, syntaxFacts, cancellationToken, findInsideTrivia: true).ConfigureAwait(false);
-            return token;
+                _refactorNotifyServices, symbolicInfo, _globalOptions.CreateProvider(), cancellationToken);
```

Comment:
the moved code doesn't deal with editor concepts like the notification services.  that's all still kept here.